Adding promote, repeat, remove, playnow commands. Adds locally cached thumbnail when the currently playing song is displayed.

Artanys opened the pull request:
promote: Promotes last song in queue to front
promote [position]: Promotes song at position to front

repeat: Cycles between repeat all, repeat single, or repeat none

remove: Removes 1st song in queue
remove [position]: Removes song at position

playnow: Immediately plays the requested song, skipping whatever is currently playing (does not work for playlists)

Adds local cache of thumbnails that are uploaded with the various now playing messages

imayhaveborkedit committed:
Update 1.9.5_2 (#390) (#391)

#390

imayhaveborkedit committed:
Don't mind me

imayhaveborkedit committed:
ffs pip

imayhaveborkedit committed:
This is THE WORST SHIT

Artanys committed:
Adding promote command, precursor to repeat command

Artanys committed:
Repeat command

imayhaveborkedit committed:
Fix for discord.py breaking change

Artanys committed:
Adding 'remove' command

Artanys committed:
Adding shortened thumbnail image urls to song information

Uses https://tny.im simple url shortener API to shorten thumbnails urls,
but falls back to long thumbnail url if shortener not available. I
wanted to have the thumbnail link embedded within the song title, but
discord does not allow custom hyperlinks

Artanys committed:
Use aiohttp for thumbnail url shortener

Artanys committed:
Add playnow command

adds a new song to the queue, promotes it to the front, then skips the
currently playing song

also fixed up a skip issue when in repeat single mode

Artanys committed:
Adding local cache of thumbnails, instead of using url shortener

Artanys committed:
Removing thumbnail URL shortening

Artanys committed:
Fixed OS specific pathing

Artanys committed:
fixing setavatar when no url or image given

Artanys committed:
Merge remote-tracking branch 'refs/remotes/SexualRhinoceros/master' into dev

Artanys committed:
update gitignore for vs

Artanys committed:
adding del fuc with session closing

Artanys committed:
forgot aiosession close

Artanys committed:
add config for optional thumbnails

jayktaylor reviewed (changes requested):
Resolve merge conflicts

Artanys committed:
dice roller

Adds a roll command for generating random values. Roll one die with roll
[MAXROLL] or multiple dice with [1-100]d[MAXROLL].

Artanys committed:
bug in numDice

wouldnt work if you roll 1 dice e.g. 1d20

MattBSG reviewed (changes requested):
If this is updated with the current version of the review branch then we can look to move forward with this. Otherwise this is so outdated with the last commit in 2016 that I'll close this if there is no activity soon since it is going to be a battle to pull a lot of this to the new code base and resolves all the conflicts.
